src/main/java/com/example/bookshop/service/impl/BookServiceImpl.java

```diff
@@ -103,4 +88,24 @@ public List<BookDtoWithoutCategoryIds> getBooksByCategoryId(Long id) {
                 .map(bookMapper::toBookDtoWithoutCategories)
                 .toList();
     }
+
+    private void setCategoriesToCreatedBook(Book book, CreateBookRequestDto bookRequestDto) {
+        Set<Category> categories = bookRequestDto.getCategoryIds().stream()
+                .map(c -> categoryRepository.findById(c).orElseThrow(
```

Comment:
You did not resolve mentor's comment regarding efficiency. It is very inefficient to call DB such way. Better call this repository method `List<T> findAllById(Iterable<ID> ids);` to make only one DB call.